qiskit/circuit/classical/expr/__init__.py

```diff
@@ -160,6 +160,11 @@
 suitable "key" functions to do the comparison.
 
 .. autofunction:: structurally_equivalent
+
+Some expressions have associated memory locations with them, and some may be purely temporaries.
+You can use :func:`is_lvalue` to determine whether an expression has such a memory backing.
```

Comment:
Maybe reword this slightly like:

```suggestion
Some expression have an associated memory location, and others may be purely temporary.
You can use :func:`is_lvalue` to determine whether an expression has an associated memory
location.
```

Reply:
Done in bbb2b406d6.